Assets/Scripts/Game/MagicAndEffects/Effects/Mysticism/Teleport.cs

```diff
@@ -231,15 +223,6 @@ private void PlayerEnterExit_OnRespawnerComplete()
             if (caster == null || caster != GameManager.Instance.PlayerEntityBehaviour)
                 return;
 
-            // Get peered SerializablePlayer and PlayerEnterExit
```

Comment:
This does seem redundant here - but I think I added this for an edge case. I just can't recall exactly the situation right now. I think the event can fire out of order to effect starting, which means these references are not otherwise collected.

Rather than remove this completely, I'd prefer a CacheReferences() call in place of this block. It would also be helpful if CacheReferences() returned true when both references are successfully collected. Then in place of this block we could use:

    if (!CacheReferences())
        return;

That would make me feel a bit better when handling this event.